### PR TITLE
Group classes by function or type

### PR DESCRIPTION
**Pull request by neomatrix369**
On the back of #194 I have started to slowly group together classes, methods/functions, fields/variables that come across as fulfilling similar functionality.

For this sweep I picked these 3 categories:

- Language support
- OS/Arch/Platform 
- Suite

Doing this should help identify temporal coupling, preparing the file for refactoring later on.

I re-ran the graal build via https://circleci.com/gh/neomatrix369/awesome-graal/503 which builds this particular repo and branch.

Build status of this branch: [CircleCI](https://circleci.com/gh/neomatrix369/awesome-graal/tree/build-mx-from-another-repo)

### CODE REVIEW

File: run-pylint-check.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (dougxc) [inline run-pylint-check.sh]: Why do we need this file?

**Comment by dougxc** [comment]
Given the size of the diff with all this rearranging, please make sure the PR contains *only* rearranging so that it should be trivially equivalent to the original code. Note that there's still a risk in breakage when moving module level statements (which includes class and method definitions) but hopefully our gate with catch any problems related to that.

**Comment by neomatrix369** [comment]
> Given the size of the diff with all this rearranging, please make sure the PR contains _only_ rearranging so that it should be trivially equivalent to the original code. Note that there's still a risk in breakage when moving module level statements (which includes class and method definitions) but hopefully our gate with catch any problems related to that.

Yes I have taken care to see there is no reason for breakage - the IDE is helping, plus I'm building using the build script to run the changed version of mx each time. These changes are just moving the blocks closer to each other so they can be extracted.

There is still a chance of breakage, and this could indicate to us something when it happens either code that needs fixing, usually less optimal couplings can cause it.

**Comment by neomatrix369** [comment]
I'll rebase mx.py to fix the conflict reported by github.

**Comment by dougxc** [comment]
Please  don't rebase. If I cannot resolve conflicts when integrating, I'll reach out for help. Please let me know once you're ready for me to try integrate this PR.

**Comment by neomatrix369** [comment]
Please proceed, this PR is good to go!

File: run-pylint-check.sh (hunk 1 above)
Comment (neomatrix369) [reply run-pylint-check.sh]: To enable running checks done by this command, before creating a PR

Comment (dougxc) [reply run-pylint-check.sh]: Ok but I don't think we need it in the repo. Can you please remove it.

Comment (neomatrix369) [reply run-pylint-check.sh]: Done! Removed!

**Comment by neomatrix369** [comment]
@dougxc any chance of this being reviewed if you have the bandwidth?

**Comment by dougxc** [comment]
I'm trying to integrate it now.

**Comment by gilles-duboscq** [comment]
For anybody ever needing to review this kind of changes, this was reviewed using [this script](https://gist.github.com/gilles-duboscq/b4edd46359ca71db268c23e0a4a93b0b#file-cut_mx-py) based on an idea from @ansalond.
This cuts the top-level declarations of a python script into files that can then easily be processed by `diff`.

**Comment by neomatrix369** [comment]
> For anybody ever needing to review this kind of changes, this was reviewed using [this script](https://gist.github.com/gilles-duboscq/b4edd46359ca71db268c23e0a4a93b0b#file-cut_mx-py) based on an idea from @ansalond.
> This cuts the top-level declarations of a python script into files that can then easily be processed by `diff`.

Thanks for letting us know.

Let's hope we don't have to do reviews on such file for long!

**Comment by neomatrix369** [comment]
> I'm trying to integrate it now.

Thanks!

**Comment by neomatrix369** [comment]
> For anybody ever needing to review this kind of changes, this was reviewed using [this script](https://gist.github.com/gilles-duboscq/b4edd46359ca71db268c23e0a4a93b0b#file-cut_mx-py) based on an idea from @ansalond.
> This cuts the top-level declarations of a python script into files that can then easily be processed by `diff`.

Good work with the script - nice work! I just applied it to the PR #197

**Comment by neomatrix369** [comment]
> For anybody ever needing to review this kind of changes, this was reviewed using [this script](https://gist.github.com/gilles-duboscq/b4edd46359ca71db268c23e0a4a93b0b#file-cut_mx-py) based on an idea from @ansalond.
> This cuts the top-level declarations of a python script into files that can then easily be processed by `diff`.

I have created a bash script which can be run after running your `cut.py` script, see https://gist.github.com/neomatrix369/a945d9766ccedceadf37b97e467eeab7

Using `diff` it gathers the files that are different and captures the differences which can later be posted in a diff.

**Comment by gilles-duboscq** [comment]
> I think the `cut.py` script should be made part of the python repos like `mx` under some folder i.e. `dev-tools` or `dev-utils` so everyone can use it for reviewing large source files.

I think it is (and should remain!) rather rare to make changes that shuffle everything around. It is not about reviewing large source files (the standard tools do that just fine), it's about reviewing changes which just re-order a lot of things (regardless of the size of the files).

**Pull request merged**